src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -40,10 +40,22 @@ public override RefKind RefKind
         {
             get
             {
-                return
-                    ((object)_containingType == null || _containingType.TypeKind != TypeKind.Struct) ? RefKind.None :
-                    ((object)_containingMethod != null && _containingMethod.MethodKind == MethodKind.Constructor) ? RefKind.Out :
-                    RefKind.Ref;
+                if (ContainingType?.TypeKind != TypeKind.Struct)
+                {
+                    return RefKind.None;
+                }
+
+                if (_containingMethod?.MethodKind == MethodKind.Constructor)
```

Comment:
The RefKind of "this" is used only in data flow analysis and that only cares about it being "Out" in a struct ctor and not "Out" in regular members.

Basically the change makes it to return a correct value (outside of the constructor), but none of the callers would seem to care about that.


---
In reply to: [118565720](https://github.com/dotnet/roslyn/pull/19713#discussion_r118565720) [](ancestors = 118565720,118552148)